sys/posix/sockets/posix_sockets.c

```diff
@@ -467,6 +463,30 @@ int connect(int socket, const struct sockaddr *address, socklen_t address_len)
     switch (s->type) {
 #ifdef MODULE_CONN_TCP
         case SOCK_STREAM:
+            /* "If the socket has not already been bound to a local address,
```

Comment:
this looks a lot like that what you introduced for `SOCK_DGRAM`.
Could you refactor this out of the switch? It seems to be socket type agnostic.